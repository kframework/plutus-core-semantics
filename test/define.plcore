(program

(module Foo (imported) (exported () ()) (local () ())
  (define four (builtin addInteger 2 2))

  (define result (builtin multiplyInteger Foo.four Foo.four))
)

)
