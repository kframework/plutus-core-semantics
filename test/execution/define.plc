(program

(module Foo (import) (export () ())
  (declare four (integer))
  (define four (builtin addInteger 2 2))

  (define result (builtin multiplyInteger Foo.four Foo.four))
)

)
