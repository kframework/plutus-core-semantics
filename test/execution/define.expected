<T>
  <k> . </k>
  <lenv> .Map </lenv>
  <genv> ( Foo . result ) |-> 16 ( Foo . four ) |-> 4 </genv>
  <modules>
    <module>
      <curr> 0 </curr>
      <name> Foo </name>
      <data> .TyExpList </data>
      <decl> four  .NameList </decl>
      <import> .ModNmList </import>
      <exportTy> .TyExpList </exportTy>
      <exportTm> .NameList </exportTm>
    </module>
  </modules>
</T>
